### PR TITLE
external-api: external-match: Allow quote denominated external orders

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR allows for quote-denominated external orders at the api layer. We do this by sampling a price and converting a quote-denominated order to a base-denominated one before passing the order to the matching engine.

I also refactored some logic in the price reporter's queue to give a cleaner interface.

### Testing
- Unit tests pass
- Tested with both quote denominated and base denominated orders

### CODE REVIEW

File: external-api/src/http/external_match.rs [hunk 1]
```diff
@@ -65,28 +66,48 @@ pub struct ExternalOrder {
...
     /// The side of the market this order is on
     pub side: OrderSide,
-    /// The amount of the order
-    pub amount: Amount,
+    /// The base amount of the order
+    #[serde(default, alias = "amount")]
+    pub base_amount: Amount,
+    /// The quote amount of the order
+    #[serde(default)]
+    pub quote_amount: Amount,
```
Comment (sehyunc) [inline external-api/src/http/external_match.rs:74]: We should explicitly check for one of these fields to exist and return an error if not.

Comment (sehyunc) [reply external-api/src/http/external_match.rs:74]: Should also error if both exist, since then it's ambiguous what to set as the `base_amount`.

**Review by sehyunc — APPROVED** [review]
LGTM, would be nice to have helpful error messages when creating an external order if none or both amounts are provided

File: external-api/src/http/external_match.rs (hunk 1 above)
Comment (joeykraut) [reply external-api/src/http/external_match.rs:74]: done

**Pull request merged**